### ISSUE TITLE
Add method to manually interrupt the agent.

### ISSUE THREAD

**Issue by martin-purplefish** [livekit/agents#1179]
It would be nice to have more control over the agent from the outside.

One use case I have: The user hangs up while the agent is speaking. I would like to be able to manually interrupt the agent, so that I can log what the agent had already said on my backend.
 
Basically, in my participant_disconnected handler, I'd like to be able to say `agent.interrupt()` - which should interrupt uninterruptable speech - and then flush the final events so I can send them to my backend.

**Comment by longcw** [comment]
@martin-purplefish Is https://github.com/livekit/agents/pull/1294 works for you? we can close this one if so.

**Comment by martin-purplefish** [comment]
Yep! Though I need something for the human side as well - basically flush when I hang up. But I'll close this one - THANK YOU!

**Comment by Jeandcc** [comment]
Hi folks. 

What do you suggest in terms of approach to "resume" the agent? Say that we call `agent.interrupt` because the user started to draw something on screen. How could I "resume" the agent once I determine it's time for the agent to go back speaking?

I don't love the idea of using `agent.say` because it requires adding an extra interaction that is not needed. Ideally, the agent would just pick up from where it left off.

@martin-purplefish , @longcw

**Comment by longcw** [comment]
@Jeandcc you can use `session.generate_reply()` in the agents 1.0

**Comment by Jeandcc** [comment]
Hi @longcw ! Forgot to send this to you, but thank you for the reply! Will use it after I rework my codebase to work with the v1.
